### DISCUSSION TITLE
Help: Moved all markers to history

### DISCUSSION THREAD

**Discussion by vmicho** [osmandapp/OsmAnd#16427] [Q&A]
Is there a reasonable way to undo the 'move all to history' on markers?
I can eventually restore one-by-one, but this suxx completely. The history screen is trash. I See lots of markers that I've already marked as 'Passed' even long time ago.

The only data shown on marker from history is 'Last used', whatever that means - definitely not a creation date as I see some markers as used today, but I am pretty sure I used them half year ago. Also there is no way to tell if the markers are Passed, from the history
Some kind of filter to show only un-passed and restore all shown would definitely be helpful.

The only idea I have right now would be to restore and process today's markers (I put notes on map what to edit in OSM), mark them passed, and import yesterday's exported markers -> but this also will not restore them as flags on map, rather like some favorites style (not to my taste).
:(

**Comment by vshcherb** [comment]
I'm afraid not, if markers are grouped in GPX it's probably possible to change metadata inside GPX itself

**Comment by vmicho** [comment]
Well looks like the usage date was same for all that were moved to history now, so I just restored one by one, I had over 200 markers.
Funny thing is that the app crashed somewhere over 200 restored markers :0
